collector/collector.go

```diff
@@ -9,23 +10,37 @@ import (
 	"github.com/stmcginnis/gofish"
 )
 
+// ContextAwareCollector is mostly a prometheus.Collector, but cannot satisfy the Collector
+// interface as it accepts an input context.Context.
+// See https://github.com/prometheus/client_golang/issues/1538 for an upstream proposal
+// that would be similar to what this interface means to achieve.
+type ContextAwareCollector interface {
+	// Collect satisfies traditional prometheus.Collector semantics; it uses a context.TODO as a parent
+	Collect(ch chan<- prometheus.Metric)
```

Comment:
Do we need to keep `Collect` function? For a backward compatibility? I'd probably break (I'm not sure how much we keep the compatibiilty already compared to an upstream)